internal/gatewayapi/route.go

```diff
@@ -719,6 +723,28 @@ func (t *Translator) processHTTPRouteParentRefListener(route RouteContext, route
 	return hasHostnameIntersection
 }
 
+func buildRouteMetadata(route RouteContext) *ir.ResourceMetadata {
+	return &ir.ResourceMetadata{
+		Kind:        route.GetObjectKind().GroupVersionKind().Kind,
+		Name:        route.GetName(),
+		Namespace:   route.GetNamespace(),
+		Annotations: filterEGPrefix(route.GetAnnotations()),
+	}
+}
+
+func filterEGPrefix(in map[string]string) map[string]string {
+	out := map[string]string{}
+	for k, v := range in {
+		if strings.HasPrefix(k, egPrefix) {
+			out[strings.TrimPrefix(k, egPrefix)] = v
```

Comment:
should we trim the prefix or let it be ?

Reply:
It feels a bit redundant to me, since only annotations with this prefix will be included.

Reply:
cool, lets leave as is